packages/driver/cypress/integration/e2e/multi-domain/commands/multi_domain_screenshot.spec.ts

```diff
@@ -0,0 +1,183 @@
+      const automationStub = cy.stub(Cypress, 'automation').withArgs('take:screenshot').resolves(serverResult)
+
+      cy.screenshot({ capture: 'runner' })
+      .then(() => {
+        expect(automationStub).to.be.calledOnce
+        expect(automationStub.args[0][1].capture).to.equal('runner')
+      })
+    })
+  })
+
+  it('captures the viewport', () => {
+    cy.viewport(600, 200)
+
+    cy.switchToDomain('foobar.com', [this.serverResult], ([serverResult]) => {
+      const automationStub = cy.stub(Cypress, 'automation').withArgs('take:screenshot').resolves(serverResult)
+
+      cy.screenshot({ capture: 'viewport' })
+      .then(() => {
+        expect(automationStub).to.be.calledOnce
+        expect(automationStub.args[0][1].capture).to.equal('viewport')
+      })
+    })
+  })
+
+  it('supports multiple titles', () => {
+    cy.switchToDomain('foobar.com', [this.serverResult], ([serverResult]) => {
+      const automationStub = cy.stub(Cypress, 'automation').withArgs('take:screenshot').resolves(serverResult)
+
+      cy.screenshot()
+      .then(() => {
+        expect(automationStub.args[0][1].titles).to.deep.equal(['screenshot specs', 'supports multiple titles'])
+      })
+    })
+  })
+
+  // FIXME: Add support for blackout option. Has cross-domain issue due to the blackout logic
+  // being called from top instead of the spec bridge
+  it.skip('supports the blackout option', () => {
+    cy.switchToDomain('foobar.com', () => {
+      cy.screenshot({ blackout: ['a'] })
+    })
+  })
+
+  it('supports element screenshots', () => {
+    cy.switchToDomain('foobar.com', [this.serverResult], ([serverResult]) => {
+      const automationStub = cy.stub(Cypress, 'automation').withArgs('take:screenshot').resolves(serverResult)
+
+      cy.get('.tall-element').screenshot()
+      .then(() => {
+        expect(automationStub.args[0][1].clip.x).to.be.greaterThan(0)
+        expect(automationStub.args[0][1].clip.y).to.be.greaterThan(0)
+      })
+    })
+  })
+
+  it('supports screenshot retrying with appropriate naming', () => {
+    cy.switchToDomain('foobar.com', [this.serverResult], ([serverResult]) => {
+      const automationStub = cy.stub(Cypress, 'automation').withArgs('take:screenshot').resolves(serverResult)
+
+      cy.state('runnable')._currentRetry = 2
+
+      cy.screenshot()
+      .then(() => {
+        expect(automationStub.args[0][1].testAttemptIndex).to.equal(2)
+      })
+    })
+  })
+
+  it('calls the onBeforeScreenshot callback', () => {
+    cy.switchToDomain('foobar.com', [this.serverResult], ([serverResult]) => {
+      let called
+
+      cy.stub(Cypress, 'automation').withArgs('take:screenshot').resolves(serverResult)
+
+      cy.screenshot({
+        onBeforeScreenshot () {
+          called = true
+        },
+      })
+      .then(() => {
+        return called
+      })
+    })
+    .then((called) => {
+      expect(called).to.be.true
+    })
```

Comment:
```suggestion
      .then(() => {
        expect(called).to.be.true
      })
    })
```

Maybe I'm missing something. Any reason not to do it this way?

Reply:
Another alternative, something like:

```js
      cy.stub(Cypress, 'automation').withArgs('take:screenshot').resolves(serverResult)

      const onBeforeScreenshot = cy.stub()

      cy.screenshot({
        onBeforeScreenshot,
      })
      cy.wrap(onBeforeScreenShould).should('be.called')
``

Should give a nicer error message if it fails.